src/lib/mongoBackend/mongoSubscribeContext.cpp

```diff
@@ -133,11 +133,30 @@ HttpStatusCode mongoSubscribeContext
 
     /* Build attributes array */
     BSONArrayBuilder attrs;
-    for (unsigned int ix = 0; ix < requestP->attributeList.size(); ++ix) {
-        attrs.append(requestP->attributeList[ix]);
+    for (unsigned int ix = 0; ix < requestP->attributeList.size(); ++ix)
+    {
+      attrs.append(requestP->attributeList[ix]);
     }
     sub.append(CSUB_ATTRS, attrs.arr());
 
+
+    //
+    // StringFilter in Scope?
+    //
+    Restriction    restriction;
+    StringFilter*  stringFilterP = NULL;
+
+    for (unsigned int ix = 0; ix < requestP->restriction.scopeVector.size(); ++ix)
```

Comment:
If I'm understanding correctly, you are processing `requestP->restriction` to copy Scopes to a local `restriction` variable. Why don't just passing the `requestP->restriction` (either as pointer or as const X& as told in other comments) in the processConditionVector() call?